AmplifyPlugins/Core/AWSPluginsCoreTests/Model/GraphQLDocument/GraphQLSyncQueryTests.swift

```diff
@@ -75,4 +75,43 @@ class GraphQLSyncQueryTests: XCTestCase {
         XCTAssertEqual(document.variables["lastSync"] as? Int, 123)
     }
 
+    func testSyncGraphQLQueryForComment() {
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: Comment.self, operationType: .query)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .sync))
+        documentBuilder.add(decorator: PaginationDecorator(limit: 100, nextToken: "token"))
+        documentBuilder.add(decorator: ConflictResolutionDecorator(lastSync: 123))
```

Comment:
One thing I noticed is odd is that if i re-arrange the decorators, the unit tests break. e.g.
```
        documentBuilder.add(decorator: ConflictResolutionDecorator(lastSync: 123))
        documentBuilder.add(decorator: DirectiveNameDecorator(type: .sync))
        documentBuilder.add(decorator: PaginationDecorator(limit: 100, nextToken: "token"))
```
will not generate the `startedat` field.  I suspect that the change you made didn't impact this though.

Reply:
What's even more odd to me is that if i don't include any of the decorators, the queryDocument looks wildly different.  I'm not super familiar with this portion of datastore -- is this expected?

Reply:
yes the decorators are not that smart per-say, you have to arrange them in the correct order to achieve what you want. there's inheritly a dependency chain that's not enforced to allow us to iterate on this faster as we're still figuring out what we need to support. For the 80% use case, there are extensions to create `GraphQLRequest` here https://github.com/aws-amplify/amplify-ios/tree/master/AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest so in most cases, the developer does not need to arrange the decorators in the right way.

Reply:
DataStore will use these https://github.com/aws-amplify/amplify-ios/blob/master/AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest/GraphQLRequest%2BAnyModelWithSync.swift#L61